GitCommands/Git/Executable.cs

```diff
@@ -91,22 +90,23 @@ public ProcessWrapper(string fileName, string arguments, string workDir, bool cr
                     }
                 };
 
-                _logOperation = CommandLog.LogProcessStart(fileName, arguments);
-
                 _process.Exited += OnProcessExit;
-
                 _process.Start();
 
+                _logOperation = CommandLog.LogProcessStart(fileName, arguments);
```

Comment:
I'd avoid moving this line down. It exposes a different race where `OnProcessExit` tries to use a `_logOperation` before it's defined.